firedrake/checkpointing.py

```diff
@@ -806,8 +931,11 @@ def save_function(self, f, idx=None, name=None):
                 path = os.path.join(base_path, str(i))
                 self.require_group(path)
                 self.set_attr(path, PREFIX + "_function", fsub.name())
-                self.save_function(fsub, idx=idx)
+                self.save_function(fsub, idx=idx, t=t, timestep=timestep)
             self._update_mixed_function_name_mixed_function_space_name_map(mesh.name, {f.name(): V_name})
+            if idx is not None:
+                self._set_timestep(mesh=mesh, name=f.name(),
+                                   idx=idx, t=t, timestep=timestep)
```

Comment:
```suggestion
```
To avoid duplication, I think it is better if we just store timestepping information directly on components. When we `get_timesteps` for a mixed function, we can just grab info from component 0.